src/engine/context.rs

```diff
@@ -0,0 +1,67 @@
+use crate::error::{bail, Result};
+use crate::parser::ast::*;
+use serde_json::Value;
+
+pub struct Context {
```

Comment:
Is is not immediately clear what the context may be used for just by looking at the name/definition. I gather that probably it's a flying parameter, used to pass evaluation state ? Would a bit of documentation here help ?

Reply:
For now, it's just a _placeholder_ for context data. The plan is to include more things here and it will make it much more clearer. Especially when #9 will be implemented.